demo/console.php

```diff
@@ -1,27 +1,50 @@
 <?php declare(strict_types=1);
 
-use Silly\Application;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 
 require __DIR__.'/../vendor/autoload.php';
 
-$silly = new Application;
+$app = new Application();
 
-$silly->command('hello [name]', function (string $name = 'World!', OutputInterface $output) {
-    $output->writeln('Hello ' . $name);
-});
-$silly->command('phpinfo', function (OutputInterface $output) {
-    ob_start();
-    phpinfo();
-    $phpinfo = ob_get_clean();
-    $output->write($phpinfo);
-});
-$silly->command('error', function (OutputInterface $output) {
-    $output->writeln('There was an error!');
-    return 1;
-});
-$silly->command('sleep', function () {
-    sleep(120);
+$app->add(new class('hello') extends Command {
```

Comment:
The `register` method should be used for this case.

```php
$app->register('hello')
    ->addArgument('name', InputArgument::OPTIONAL, '', 'World!')
    ->setCode(function (InputInterface $input, OutputInterface $output): int {
        $output->writeln('Hello, ' . $input->getArgument('name'));
        
        return Command::SUCCESS;
    });
```

Reply:
Way better, thanks!